### ISSUE TITLE
Incorrect scrolling of UITextView inside ScrollView

### ISSUE THREAD

**Issue by FranDepascuali** [facebook/react-native#20298]
## Environment:
  OS: macOS High Sierra 10.13.4
  Node: 10.0.0
  Yarn: 1.6.0
  npm: 6.0.0
  Watchman: 4.9.0
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: Not Found

**Packages: (wanted => installed)**
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.1
  react-native: 0.56 => 0.56

## Description
I'm using a native `UITextView` exported to `React Native` because there where some things (like rich text) that we couldn't do with `TextInput`. The `UITextView` is a subview of other `ScrollView`. 
We are facing a bug in which when the user presses backspace (and other keys, but it consistently happens with backspace) the parent `ScrollView` of the `UITextView` is scrolled incorrectly.
Check visually:
![bug_rn_scroll_to_rect_2](https://user-images.githubusercontent.com/12101394/42962993-68471a16-8b69-11e8-9ced-2890f732327d.gif)

We could track that this originates in this https://github.com/facebook/react-native/commit/fa1d4e8 commit. 

<img width="525" alt="screen shot 2018-07-19 at 16 00 37" src="https://user-images.githubusercontent.com/12101394/42964293-f53fa110-8b6c-11e8-8017-744da03921ee.png">

When nothing was done in `scrollRectToVisible`, we didn't have this bug.
I tried a lot to find the root cause of this bug, but with no success. I could see via a breakpoint that each time a new keystroke is pressed, `scrollRectToVisible` is eventually called (check the following stack frame).
<img width="501" alt="screen shot 2018-07-19 at 15 42 12" src="https://user-images.githubusercontent.com/12101394/42963468-add5b1fe-8b6a-11e8-8cab-36a3a0b73909.png">

There is no place in that code where we called directly `scrollRectToVisible`, but it is called internally by `UITextView` when a new key is pressed. **Note that it is called in the parent scrollView that contains the UITextView, and it receives by parameter the rect of the UITextView**

From Apple's documentation on `scrollRectToVisible`:
```Scrolls a specific area of the content so that it is visible in the receiver.
This method scrolls the content view so that the area defined by rect is just visible inside the scroll view. If the area is already visible, the method does nothing.
```

So for any reason `scrollRectToVisible` is scrolling the parent scrollview because `UITextView` is not displayed. However, it was in fact displayed.

I also took screenshots of the state of `scrollRectToVisible` when a backspace key was pressed:
**Before pressing backspace**
<img width="941" alt="screen shot 2018-07-19 at 12 12 50" src="https://user-images.githubusercontent.com/12101394/42964132-8895f640-8b6c-11e8-908c-ed4729989211.png">

**After pressing backspace**
<img width="901" alt="screen shot 2018-07-19 at 12 13 06" src="https://user-images.githubusercontent.com/12101394/42964141-8e01636c-8b6c-11e8-86c4-7bd989030bf6.png">

So the parameters don't actually change except the width of the `UITextView`.

## Steps to reproduce
I tried extracting this to a small project, but I couldn't. I created a `UITextView` exported to RN and had a `ScrollView` with that child component. The thing is that in our code we do some things like zooming which makes it difficult to extract all the code that is affecting the view.

So basically, I'm looking for any further help or direction to go. 

Thank you very much!

**Comment by neiker** [comment]
Same problem here and I can't found the solution so I have tried to swizzled `scrollRectToVisible` to a noop but I have not success so I had to forked RN.
Will be really nice to expose a way disable this "magical" behabior using a prop from javascript since may not be desired in some cases.
